service/client_state.cc

```diff
@@ -56,11 +56,18 @@ void service::client_state::validate_login() const {
 
 void service::client_state::ensure_not_anonymous() const {
     validate_login();
+    if (is_from_maintenance_socket()) {
+        return;
+    }
     if (auth::is_anonymous(*_user)) {
         throw exceptions::unauthorized_exception("You have to be logged in and not anonymous to perform this request");
     }
 }
 
+bool service::client_state::is_from_maintenance_socket() const {
+    return _auth_service->is_used_by_maintenance_socket();
```

Comment:
I don't understand what I'm seeing here:
Please correct me if I'm wrong, but auth_service has "is_used_by_maintenance_socket" if **also** used through maintenance socket. So how can we assume that some other socket's request is "from maintenance socket" and therefore "not anonymous"?
Unless I'm misunderstanding how this works, this looks incorrect to me.

If the "auth service" used by the maintenance socket is a separate "auth service" used only by it, please add a comment to is_used_by_maintenance_socket() clarifying this situation - that it means it is used **only** by a maintenance socket. Maybe it should even be named accordingly is_maintenance_socket() or for_maintenance_socket() - emphasizing that this object is only for the maintenance socket and nothing else.